rend3/src/managers/skeleton.rs

```diff
@@ -91,13 +100,21 @@ impl SkeletonManager {
         self.registry.insert(handle, internal);
     }
 
-    pub fn ready(&mut self) {
+    pub fn ready(&mut self, mesh_manager: &mut MeshManager) {
         profiling::scope!("Skeleton Manager Ready");
-        self.registry.remove_all_dead(|_, _, _| {});
+        self.registry.remove_all_dead(|_, handle_idx, skeleton| {
+            self.global_joint_count -= skeleton.joint_matrices.len();
+
+            // Clean back references in the mesh data
+            let mesh = mesh_manager.internal_data_mut(skeleton.mesh_handle.get_raw());
+            let index = mesh.skeletons.iter().position(|sk| sk.idx == handle_idx).unwrap();
+            mesh.skeletons.swap_remove(index);
+        });
     }
 
     pub fn set_joint_matrices(&mut self, handle: RawSkeletonHandle, joint_matrices: Vec<Mat4>) {
         let skeleton = self.registry.get_mut(handle);
+        assert_eq!(joint_matrices.len(), skeleton.joint_matrices.len());
```

Comment:
Let's add a error message here of some kind:
```suggestion
        assert_eq!(joint_matrices.len(), skeleton.joint_matrices.len(), "Joint matrix count cannot be changed");
```

Not sure that's the best message, but you get the picture.